### PR TITLE
Sync v2-5-stable with v2-5-test to release 2.5.0 

### PR DESCRIPTION
**Pull request by ephraimbuddy**
Time for `2.5.0rc1`!

### CODE REVIEW

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Very cool release - no dramatic changes, but steady stream of improvements: :muscle:

File: README.md [hunk 1]
```diff
@@ -86,7 +86,7 @@ Airflow is not a streaming solution, but it is often used to process real-time d
 
 Apache Airflow is tested with:
 
-|                     | Main version (dev)           | Stable version (2.4.2)       |
+|                     | Main version (dev)           | Stable version (2.5.0)       |
```
Comment (raphaelauv) [inline README.md:89]: missing PostgreSQL 15 support for 2.5.0

Comment (potiuk) [reply README.md:89]: good catch

File: RELEASE_NOTES.rst [hunk 2]
```diff
@@ -21,6 +21,279 @@
 
 .. towncrier release notes start
 
+Airflow 2.5.0 (2022-11-28)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]:
```suggestion
Airflow 2.5.0 (2022-11-30)
```

File: RELEASE_NOTES.rst [hunk 3]
```diff
@@ -21,6 +21,279 @@
...
+--------------------------
+
+Significant Changes
+^^^^^^^^^^^^^^^^^^^
+
+- ``airflow dags test`` no longer performs a backfill job.
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]:
```suggestion
``airflow dags test`` no longer performs a backfill job (#26400)
""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
```

File: RELEASE_NOTES.rst [hunk 4]
```diff
@@ -21,6 +21,279 @@
...
+
+  In order to make ``airflow dags test`` more useful as a testing and debugging tool, we no
+  longer run a backfill job and instead run a "local task runner". Users can still backfill
+  their DAGs using the ``airflow dags backfill`` command. (#26400)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]:
```suggestion
  their DAGs using the ``airflow dags backfill`` command.
```

File: RELEASE_NOTES.rst [hunk 5]
```diff
@@ -21,6 +21,279 @@
...
+- Airflow config section ``kubernetes`` renamed to ``kubernetes_executor``
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]:
```suggestion
Airflow config section ``kubernetes`` renamed to ``kubernetes_executor`` (#26873)
"""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
```

File: RELEASE_NOTES.rst [hunk 6]
```diff
@@ -21,6 +21,279 @@
...
+
+  KubernetesPodOperator no longer considers any core kubernetes config params, so this section now only applies to kubernetes executor. Renaming it reduces potential for confusion. (#26873)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]:
```suggestion
  KubernetesPodOperator no longer considers any core kubernetes config params, so this section now only applies to kubernetes executor. Renaming it reduces potential for confusion.
```

File: RELEASE_NOTES.rst [hunk 7]
```diff
@@ -21,6 +21,279 @@
...
+- ``ExternalTaskSensor`` no longer hangs indefinitely when ``failed_states`` is set, an ``execute_date_fn`` is used, and some but not all of the dependent tasks fail. Instead, an ``AirflowException`` is thrown as soon as any of the dependent tasks fail.
+
+  Any code handling this failure in addition to timeouts should move to cathing the ``AirflowException`` baseclass and not only the ``AirflowSensorTimeout`` subclass. (#27190)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]: This will need a similar change as above, but we probably need a shorter title for this?

File: RELEASE_NOTES.rst [hunk 8]
```diff
@@ -21,6 +21,279 @@
...
+- Update index.rst (#27529)
+- Fix wording in DAG Runs description(#27470)
+- Add note about pushing the lazy XCom proxy to XCom (#27250)
+- Document that KubernetesExecutor overwrites container args (#27450)
+- Fix BaseOperator link (#27441)
+- [docs] best-practices add use variable with template example. (#27316)
+- docs for custom view using plugin (#27244)
+- Update SLA wording to reflect it is relative to Dag Run start. (#27111)
+- Add missed import in the Trigger Rules example (#27309)
+- Update dags.rst (#27110)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]: Typo, can skip this one.
```suggestion
```

File: RELEASE_NOTES.rst [hunk 9]
```diff
@@ -21,6 +21,279 @@
...
+- Fix typo in ``set-up-database.rst`` (#27137)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]:
```suggestion
```

File: RELEASE_NOTES.rst [hunk 10]
```diff
@@ -21,6 +21,279 @@
...
+- set executor.job_id to BackfillJob.id for backfills (#27020)
+- Fix some bug in web ui dags list page (auto-refresh & jump search null state) (#27141)
+- Listener: Set task on sqlalchemy taskinstance object (#27167)
+- fix yarn warning from d3-color (#27139)
+- Don't overwrite connection extra with invalid json (#27142)
+- Simplify origin string cleaning (#27143)
+- Don't re-patch pods that are already controlled by current worker (#26778)
+- Add separate error handler for 405(Method not allowed) errors (#26880)
+- Avoid 500 on dag redirect (#27064)
+- Make tracebacks opt-in (#27059)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]:
```suggestion
```

Hmm, these were already in 2.4.2? Might be others as well.

Comment (ephraimbuddy) [reply RELEASE_NOTES.rst]: Yeah. The changelog command was getting everything since 2.4.1. I have manually sorted the bugfixes

**Review by kaxil — APPROVED** [review]

**Pull request merged**